### PR TITLE
Fix the author of the comment also gets mail

### LINKED ISSUE #407: User gets an email notification for his own comment
## I'm submitting a...

- [x] Bug report  
- [ ] Regression (a behavior that used to work and stopped working in a new release)
- [ ] Feature request
- [ ] Documentation issue or request

## Context
When writing a comment in a repository where the logged in user is an admin or author, that user gets an email notification notifying him of the comment he just wrote.

### PR DESCRIPTION
**Pull request by BaskovicP**
I want to resolve #407 issue: User gets an email notification for his own comment
Before making a list of recipients we filter the author of the coment. So the person who made the comment does not recieve a email but all the others in the repository do.

Please check it out: @kjuej

### CODE REVIEW

File: server/comment/hooks.js [hunk 1]
```diff
@@ -43,6 +43,9 @@ async function sendEmailNotification(comment, db) {
     author: author.profile,
     ...pick(comment, ['id', 'content', 'createdAt'])
   };
-  const recipients = map(repository.repositoryUsers, it => it.user.email);
+  const recipients = reduce(repository.repositoryUsers, (acc, it) => {
```
Comment (underscope) [inline server/comment/hooks.js]: Although this is faster consider using `lodash/without` to make it a bit more readable and compact:
```javascript
const collaborators = map(repository.repositoryUsers, 'user.email');
const recipients = without(collaborators, author.email);
```
Also what happens if there are none 😉

**Comment by BaskovicP** [comment]
Thanks @underscope ,it is much more readable like this. As _without_ always returns a new array even if we feed it something strange like undefined the array will be empty so we actually don't need to send any emails. (thats why the check for length)

**Comment by kjuej** [comment]
@BaskovicP @underscope Seems to be working as expected.

File: server/comment/hooks.js [hunk 2]
```diff
@@ -3,6 +3,7 @@
 const { broadcast, events } = require('./channel');
 const mail = require('../shared/mail');
 const map = require('lodash/map');
+const without = require('lodash/without');
```
Comment (underscope) [inline server/comment/hooks.js]: When you get a chance please order it 😉

Comment (zcuric) [reply server/comment/hooks.js]: 😉 👍

**Pull request merged**